src/input/source/led/multicolor.rs

```diff
@@ -0,0 +1,321 @@
+            .map_err(|err| Box::new(LedMcError::Write(err)))?)
+    }
+
+    /// Converts a generic dbus provided Vec<u8> to a matching Vec<u32> based on the device's
+    /// multi_index_map. This accounts for converting three u8 values into a singe Rgb or Multi
+    /// value.
+    fn dbus_write_colors(&self, colors: Vec<u8>) -> Result<(), Box<dyn Error + Send + Sync>> {
+        let contents = read_to_string(self.multi_intensity_path.as_path())
+            .map_err(|err| Box::new(LedMcError::Read(err)))?;
+        let mut values = vec![];
+        let mut visited_index = vec![];
+        for (pos, color) in self.multi_index_map.iter().enumerate() {
+            if visited_index.contains(&pos) {
+                continue;
+            }
+            match color {
+                ColorType::Multi | ColorType::Rgb => {
+                    let value: u32 = ((colors[pos] as u32) << 16u32)
+                        | ((colors[pos + 1] as u32) << 8u32)
+                        | (colors[pos + 2] as u32);
+                    values.push(value);
+                    visited_index.push(pos);
+                    visited_index.push(pos + 1);
+                    visited_index.push(pos + 2);
+                }
+                _ => {
+                    values.push(colors[pos] as u32);
+                    visited_index.push(pos);
+                }
+            }
+        }
+
+        self.write_colors(values)
+    }
+}
+
+impl Debug for LedMultiColor {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        f.debug_struct("MultiColorChassis").finish()
+    }
+}
+
+impl SourceInputDevice for LedMultiColor {
+    fn poll(&mut self) -> Result<Vec<crate::input::event::native::NativeEvent>, InputError> {
+        Ok(Vec::new())
+    }
+
+    fn get_capabilities(&self) -> Result<Vec<Capability>, InputError> {
+        Ok(Vec::new())
+    }
+}
+
+impl SourceOutputDevice for LedMultiColor {
+    fn write_event(&mut self, event: OutputEvent) -> Result<(), OutputError> {
+        log::debug!("Received output event: {event:?}");
+        match event {
+            OutputEvent::DualSense(report) => {
+                if !report.allow_led_color {
+                    return Ok(());
+                }
+                self.write_color(report.led_red, report.led_green, report.led_blue)?;
+                self.write_brightness(255)?; // Always set max brightness as DS uses only RGB.
+            }
+            _ => {
+                return Ok(());
+            }
+        }
+        Ok(())
+    }
+}
+
+/// Reads the multi_intensity path and returns an LedFixedColor with the current color setting
+fn read_color(
+    path: &Path,
+    index: &[ColorType],
+) -> Result<Vec<HashMap<ColorType, u8>>, Box<dyn Error + Send + Sync>> {
+    let mut current_color: Vec<HashMap<ColorType, u8>> = Vec::new();
+
+    let color_contents = read_to_string(path).map_err(|err| Box::new(LedMcError::Read(err)))?;
```

Comment:
If we have defined error types, then we should use that in the result instead of a `Box<dyn Error>`:

```suggestion
fn read_color(path: &Path, index: &[ColorType]) -> Result<Vec<HashMap<ColorType, u8>>, LedMcError> {
    let mut current_color: Vec<HashMap<ColorType, u8>> = Vec::new();

    let color_contents = read_to_string(path).map_err(|err| LedMcError::Read(err))?;
```